.github/workflows/deploy_release_to_marketplace.yml

```diff
@@ -0,0 +1,57 @@
+name: autokittehVSCode CI - Publish to VSCode Marketplace
+
+on:
+  workflow_dispatch:
+    inputs:
+      release-type:
+        type: choice
+        description: "Release type"
+        options:
+          - "pre-release"
+          - "release"
+        required: true
+      tag:
+        type: string
+        description: "Git tag to publish"
+        required: true
+
+jobs:
+  publish-pre-release-to-marketplace:
+    name: Publish Pre-Release to VSCode Marketplace
+    if: github.event.inputs.release-type == 'pre-release'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.tag }}
+          token: ${{ secrets.PAT_TOKEN }}
+          submodules: true
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "21.2.0"
+          cache: "npm"
+
+      - name: Build from scratch pre-release extension version
+        run: |
+          git submodule update --remote
+          npm ci
+          npm run compile:all
+          npm run build:all
+
+      - name: Publish to VSCode Marketplace as pre-release
```

Comment:
where do you set the tag if this is a pre release ?

Reply:
It takes the tag from the `package.json`